src/coreclr/jit/instr.cpp

```diff
@@ -2159,8 +2159,10 @@ void CodeGen::instGen_Set_Reg_To_Zero(emitAttr size, regNumber reg, insFlags fla
 {
 #if defined(TARGET_XARCH)
     GetEmitter()->emitIns_R_R(INS_xor, size, reg, reg);
-#elif defined(TARGET_ARMARCH)
+#elif defined(TARGET_ARM)
     GetEmitter()->emitIns_R_I(INS_mov, size, reg, 0 ARM_ARG(flags));
+#elif defined(TARGET_ARM64)
+    GetEmitter()->emitIns_Mov(INS_mov, size, reg, REG_ZR, /* canSkip */ true);
```

Comment:
This seems like a more general optimization (there are many Arm64 instructions that support `ZR` or `Immediate`). Is there something we can/should be doing in `emit` instead to automatically recognize `I == 0` and convert it to `REG_ZR` to ensure that this optimization always lights up and that devs don't need to remember to specially handle it in every scenario?

Reply:
@tannergooding I think we already do it (e.g. https://github.com/dotnet/runtime/pull/52269), just not sure how this one was skipped.

Reply:
I'll take a look r2r dump corelib to find `#0`

Reply:
That's just handling containment in lowering which isn't quite the same as something like this (which is the result of we need zero and can't contain)

I was referring to doing basically a peephole in emit that says "you're passing `#0` but this instruction also supports `REG_ZR`, so fix it up there.

Reply:
By that PR I meant that we already do it since a simple change that just enabled containing resulted in ZR everywhere and huge diffs

Reply:
I've just checked all functions in R2R'd corelib - we only use `#0` for `cmp` and `movn` now.
There are some cases where `cmp reg, #0` is followed by a branch which most likely can be folded to "branch if zero" instructions but it's not related

Reply:
`movn reg, #0` is interesting though